### PR TITLE
CASTanet Release 1: Functions without docstrings, comments, if-statements, looping constructs

### PR DESCRIPTION
**Pull request by MaddyKapfhammer**
# CASTanet Release 1: Functions without docstrings, comments, if-statements, looping constructs

## Description

This PR proposes changes to be merged into the `main` branch of CASTanet. These changes relate to all the beginning stages of the tool, including traversing a directory, creating functions to fulfill MVPs, testing and documentation. Specifically, this release contains functionality that allows a user to determine the number of comments, if-statements, looping constructs, and functions without docstrings through using a basic command line interface with commands in `main.py`. These features, including their testing, will be improved with the second release of CASTanet, including implementation of new MVPs.

### Type of Change

- [x] Feature
- [x] Testing
- [x] Documentation

## Contributors

- @MaddyKapfhammer 
- @antlet 
- @Thompsonn2 
- @BaileyMatrascia 
- @hinckleyc 

## Reminder

All GitHub Actions should be in a passing state before any pull request is merged.

### CODE REVIEW

File: castanet/counter.py
Comment (mariakimheinert) [inline castanet/counter.py:135]: This should be renamed to be more consistent with the other functions and to be more descriptive--it should start with `count_` and include `functions` somewhere in the name.

File: castanet/counter.py
Comment (mariakimheinert) [inline castanet/counter.py:78]: Might `total_loops` be a better name to be more consistent with the rest of the functions?

File: tests/test_counter.py
Comment (mariakimheinert) [inline tests/test_counter.py:17]: One good testing practice is to keep your test input as close to the tests and their assertions as possible.

This way, someone looking at a test can make sense of the test just by looking at only the test. If the test contains the input, the action under test, and the assertion, then it's easy to see why the assertion was made.

In contrast, if the test input is far away from the test (i.e. in another file), then someone would have to look at two files to make sense of a test. Here, I'd also need to look at the files in the `test_files` directory to understand why the assertion that the number of `for` loops should be 3 makes sense. Moreover, if the test input is ever updated, it's not immediately clear which tests need to be updated as well.

This is not something that needs to be _fixed_, per se, but something to think about when approaching future testing. You may also consider refactoring some of the existing code to facilitate this practice.

For example, if `count_forloops` was rewritten so that it took in a string of code, rather than a dictionary of CSTs, you could write something like:

```python
def test_count_forloops_returns_number_of_forloops():
    input = '''
for _ in range(2):
    pass
# for
'''
    assert counter.count_forloops(input) == 1
```

By looking at only this test case, I can see exactly why the count should be 1. And if I were to update the input, then the assertion that also needs to be updated is directly below and easy to remember to change.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Great work on this release! Your documentation is great and your source code works and is well-tested. Please let me know if you have any questions about the changes I've requested. In addition to the suggestions I've made in my inline feedback, please make sure to remove any `.DS_Store` files and either add `.DS_Store` to the `.gitignore` in this repository or have all team members add `.DS_Store` to their user-level `.gitignore`s.

File: castanet/counter.py
Comment (Michionlion) [inline castanet/counter.py:103]: Each `total_x` or `amount_x` function in this file has the same exact implementation (bar variable names) -- it would definitely be a good idea to consolidate them into one "sum_cast_dict` function or something similar.

File: main.py
Comment (Michionlion) [inline main.py:6]: This `help` string shows up as a description/help for your program as a whole when `main.py --help` is run. It may be more useful for users if a descriptive string about what the program does is used, as the current `help` is not related to CASTanet.

File: pyproject.toml
Comment (Michionlion) [inline pyproject.toml:11]: You may want to include the "all" extra (see the Typer [documentation](https://typer.tiangolo.com/tutorial/package/#dependencies-and-environment)). This enables some extra `Click` support, and a few other smaller features.

File: pyproject.toml
Comment (Michionlion) [inline pyproject.toml:21]: The `hello/` package should probably not be linted.

File: test_files/comments.py
Comment (Michionlion) [inline test_files/comments.py:1]: It may be a good idea to have other code (with no comments) in this file to ensure a slightly more complex Concrete Syntax Tree is created while testing.

File: pyproject.toml
Comment (Michionlion) [inline pyproject.toml:7]: You should add a name, authors, description, and other metadata for your project. Eventually you will need to publish this project on PyPI, and this metadata will be useful!

**Review by Michionlion — CHANGES_REQUESTED** [review]
This is a great start on your library (excellent name by the way, even if "Concrete Abstract" is a bit of an oxymoron)! In addition to the few bits of feedback I've given below, there are two main changes I'd like to see in your release candidate before merging it to `main` tomorrow:

First, as @mariakimheinert mentioned, there are quite a few `.DS_Store` files in the repository. These files are always a thorn in the side of any non-Mac developers, and should definitely be removed.

Second, your Typer CLI object should be registered as a script in `pyproject.toml` -- see the Poetry [documentation](https://python-poetry.org/docs/pyproject/#scripts) as well as the relevant [tutorial](https://typer.tiangolo.com/tutorial/package/) in the Typer documentation. This will make using your CLI as a user very easy, and should set you up for success when publishing your tool to PyPI.

I'm looking forward to seeing your improvements to this release, and the upcoming features you'll work on for next sprint!

File: README.md
Comment (MaddyKapfhammer) [reply README.md:27]: This has been resolved in the latest release of CASTanet.

File: README.md
Comment (MaddyKapfhammer) [reply README.md:35]: This has been corrected in latest release.

File: README.md
Comment (MaddyKapfhammer) [reply README.md:178]: This has been corrected in latest release.

File: CONTRIBUTING.md
Comment (MaddyKapfhammer) [reply CONTRIBUTING.md:12]: Thomas Antle has also agreed to being a maintainer of this tool, and his contact information has been added to `CONTRIBUTING.md`. Twitter handles have been removed and headings were changed to reflect this maintainer status.

File: test_files/comments.py
Comment (MaddyKapfhammer) [reply test_files/comments.py:1]: This will be addressed when test cases are improved upon in next sprint following @mariakimheinert's suggestions about how test cases should be structured.

File: pyproject.toml
Comment (Michionlion) [inline pyproject.toml:9]: It's generally good practice to keep executables all lowercase. Additionally, your main package is `castanet`, not `CASTanet`. Finally, `main.py`, which I think is what you're attempting to call here, is not in the `castanet` package.

File: castanet/counter.py
Comment (Michionlion) [inline castanet/counter.py:96]: This name could be more descriptive; it doesn't reflect that it counts functions without docstrings.

File: main.py
Comment (Michionlion) [inline main.py:6]: This help text should describe what your tool does.

**Review by Michionlion — CHANGES_REQUESTED** [review]
See the small bits of feedback below. Additionally, the `hello` directory should be removed; you have no need for it now.

File: main.py
Comment (Michionlion) [reply main.py:6]: This suggestion can be completed next sprint.

File: castanet/counter.py
Comment (Michionlion) [reply castanet/counter.py:96]: This suggestion can be completed next sprint.

File: pyproject.toml
Comment (mariakimheinert) [reply pyproject.toml:9]: This does not work as-is. This should be `castanet = "main:app"`.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
In addition to removing the `hello` directory as @Michionlion suggested, please make sure to update the script in `pyproject.toml`. These are the only changes I'd like to see before I give my approval.

File: pyproject.toml
Comment (MaddyKapfhammer) [reply pyproject.toml:9]: This has been resolved, closing conversation!

File: tests/test_counter.py
Comment (MaddyKapfhammer) [reply tests/test_counter.py]: Resolving until the next sprint.

**Review by Michionlion — DISMISSED** [review]

File: tests/test_counter.py
Comment (mariakimheinert) [inline tests/test_counter.py:45]:
```suggestion
    """Uses match_if_statements to identify all the if-statements in the castanet directory."""
```

**Review by mariakimheinert — APPROVED** [review]
Just made one small suggestion below. Changes look great and are good to merge!

**Review by mariakimheinert — APPROVED** [review]

**Review by Michionlion — APPROVED** [review]

**Pull request merged**
